precompiles/authorization/IICS20Authorization.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.17;
+
+import "../common/Types.sol";
+
+/// @author Evmos Team
+/// @title Authorization Interface
+/// @dev The interface through which solidity contracts will interact with smart contract approvals.
+interface IICS20Authorization {
+    /// @dev Emitted when an ICS-20 transfer authorization is granted.
+    /// @param grantee The address of the grantee.
+    /// @param granter The address of the granter.
+    /// @param allocations The Allocations authorized with this grant.
+    event IBCTransferAuthorization(
+        address indexed grantee,
+        address indexed granter,
+        Allocation[] allocations
+    );
+
+    /// @dev Approves IBC transfer with a specific amount of tokens.
+    /// @param grantee The address for which the transfer authorization is granted.
+    /// @param allocations The Allocations for the authorization.
+    function approve(address grantee, Allocation[] calldata allocations) external returns (bool approved);
+
+    /// @dev Revokes IBC transfer authorization for a specific grantee.
+    /// @param grantee The address for which the transfer authorization will be revoked.
+    function revoke(address grantee) external returns (bool revoked);
+
+    /// @dev Increase the allowance of a given grantee by a specific amount of tokens for IBC transfer methods.
+    /// @param grantee The address of the contract that is allowed to spend the granter's tokens.
+    /// @param sourcePort The port on which the packet will be sent.
+    /// @param sourceChannel The channel by which the packet will be sent.
+    /// @param denom The denomination of the Coin to be transferred to the receiver
+    /// @param amount The increase in amount of tokens that can be spent.
+    /// @return approved Is true if the operation ran successfully.
+    function increaseAllowance(
+        address grantee,
+        string calldata sourcePort,
+        string calldata sourceChannel,
+        string calldata denom,
+        uint256 amount
```

Comment:
We could potentially do this but then devs would have to build their own `Coin` type and then we would have to convert this `Coin` type into an `sdk.Coin` type because they are not directly compatible. So it's a bit more work and not sure if it's worth it at the moment as we will have to change it everywhere to stay consistent